search/search-dev-tools/search-dev-tools.php

```diff
@@ -0,0 +1,269 @@
+<?php
+/**
+ * Plugin Name: Search Dev Tools
+ * Description: Developer tools for Enterprise Search
+ * Version:     1.0.0
+ * Author:      WordPress VIP
+ * Author URI:  https://wpvip.com
+ * License:     GPLv2 or later
+ * Text Domain: vip-search
+ * Domain Path: /lang/
+ *
+ * @package Automattic\VIP\Search
+ */
+// phpcs:disable WordPress.PHP.DisallowShortTernary.Found
+namespace Automattic\VIP\Search\Dev_Tools;
+
+use Automattic\VIP\Search\Search;
+
+if ( ! ( defined( 'VIP_ENABLE_VIP_SEARCH' ) && VIP_ENABLE_VIP_SEARCH ) ) {
+	return;
+}
+
+define( 'SEARCH_DEV_TOOLS_CAP', 'edit_others_posts' );
+
+add_action( 'rest_api_init', __NAMESPACE__ . '\register_rest_routes' );
+add_action( 'admin_bar_menu', __NAMESPACE__ . '\admin_bar_node', PHP_INT_MAX );
+add_action( 'wp_enqueue_scripts', __NAMESPACE__ . '\enqueue_assets', 11 );
+add_filter( 'js_do_concat', __NAMESPACE__ . '\skip_js_do_concat', 10, 2 );
+add_action( 'wp_footer', __NAMESPACE__ . '\print_data', 5 );
+
+/**
+ * Register Dev Tools Endpoint.
+ *
+ * @return void
+ */
+function register_rest_routes() {
+	register_rest_route(
+		'vip/v1',
+		'search/dev-tools',
+		[
+			'methods'             => [
+				'POST',
+			],
+			'callback'            => __NAMESPACE__ . '\rest_callback',
+			'permission_callback' => __NAMESPACE__ . '\should_enable_search_dev_tools',
+			// Uncomment this for testing with `npm run dev`
+			// 'permission_callback' => '__return_true',
```

Comment:
Could we make this conditional to the environment we're in? Maybe using something like `WPCOM_IS_VIP_ENV` or `VIP_GO_APP_ENVIRONMENT` to figure out if we are in a dev machine?

Reply:
It's better not to introduce any differences in behavior unless absolutely needed, otherwise, we're at risk of not noticing some change that would only happen in a non-local environment. 

I've removed the note, looking back at it, it seems something for a README.